core/src/wallet/tezos/explorers/ExternalTezosLikeBlockchainExplorer.cpp

```diff
@@ -371,19 +375,45 @@ namespace ledger {
                                                                    getRPCNodeEndpoint());
         }
 
+        Future<std::shared_ptr<BigInt>>
+        ExternalTezosLikeBlockchainExplorer::getTokenBalance(const std::string& accountAddress,
+                                                             const std::string& tokenAddress) const {
+            const auto parseNumbersAsString = true;
+            return _http->GET(fmt::format("/account/mainnet/{}", accountAddress), {}, BCD_API_ENDPOINT)
+                .json(parseNumbersAsString)
+                .mapPtr<BigInt>(getContext(),
+                    [=](const HttpRequest::JsonResult &result) {
+                       const auto &json = *std::get<1>(result);
+                       if ((!json.IsObject() ||
+                           !json.HasMember("tokens") ||
+                           !json["tokens"].IsArray()) && !json.IsString()) {
+                           throw make_exception(api::ErrorCode::HTTP_ERROR,
+                                                fmt::format("Failed to get tokens for {}", accountAddress));
+                       }
+
+                       const auto tokens = json["tokens"].GetArray();
+                       for (const auto& token : tokens) {
+                           if (token["contract"].GetString() == tokenAddress) {
+                               return std::make_shared<BigInt>(BigInt::fromString(token["balance"].GetString()));
```

Comment:
Are we sure to have always "contract" and "balance" in each token?
the lib will crash if one of these 2 fields is missed

Reply:
Right now yes, but I will add HasMember(...) to prevent crashes in case there is a change in the future